main.go

```diff
@@ -62,17 +64,33 @@ func main() {
 	flag.StringVar(&logTimestampFormat, "log-timestamp-format", "millis", "log timestamp format ('millis', 'nanos', 'epoch', 'rfc3339' or 'rfc3339nano')")
 	flag.IntVar(&syncPeriodSeconds, "syncPeriodSeconds", 60*60, "The informers resync period in seconds") // default 1 hour
 	flag.StringVar(&leaderElectionResourceLock, "leader-election-resource", "configmaps", "determines which resource lock to use for leader election. option:[configmapsleases|endpointsleases|configmaps]")
+	flag.BoolVar(&ddProfilingEnabled, "ddProfilingEnabled", false, "Enable the datadog profiler")
 	logLevel := zap.LevelFlag("loglevel", zapcore.InfoLevel, "Set log level")
 
 	flag.Parse()
 
+	exitCode := 0
+	defer func() { os.Exit(exitCode) }()
+
+	if ddProfilingEnabled {
+		setupLog.Info("Starting datadog profiler")
+		if err := profiler.Start(
+			profiler.WithVersion(version.Version),
+			profiler.WithProfileTypes(profiler.CPUProfile, profiler.HeapProfile),
+		); err != nil {
+			setupLog.Error(err, "unable to start datadog profiler")
+		}
+
+		defer profiler.Stop()
+	}
```

Comment:
we should move this block under `customSetupLogging` to make sure the logger is configured properly. See the EDS fix for this https://github.com/DataDog/extendeddaemonset/pull/135